src/features/instance/browse/BrowseDataTableView.tsx

```diff
@@ -42,45 +28,102 @@ export function BrowseDataTableView() {
 		instanceId,
 		schemaName,
 		tableName,
 		hashAttribute: [''],
+		refetch: false,
 	});
 
-	const { data: searchByHashData, refetch: refetchSearchByHash } = useQuery(getSearchByHashOptions(searchByHashParams));
+	const {
+		data: searchByHashData,
+		refetch: refetchSearchByHash,
+	} = useQuery(getSearchByHashOptions(searchByHashParams));
 
 	const { dataTableColumns, hash_attribute } = formatBrowseDataTableHeader(describeTableData);
 	const [isEditModalOpen, setIsEditModalOpen] = useState(false);
-	const [sortTableDataParams, setSortTableDataParams] = useState({
-		attribute: '',
+	const [isAddModalOpen, setIsAddModalOpen] = useState(false);
+	const [{ refetch: refetchBySortTableParams, ...sortTableDataParams }, setSortTableDataParams] = useState({
+		attribute: hash_attribute,
 		descending: false,
+		refetch: false,
 	});
+	const sortingState = useMemo(() => ([{
+		desc: sortTableDataParams.descending,
+		id: sortTableDataParams.attribute,
+	}]), [sortTableDataParams]);
+
 	const [totalRecords, setTotalRecords] = useState(describeTableData.record_count);
 	const [pagination, setPagination] = useState<PaginationState>({
 		pageIndex: 0,
 		pageSize: 20,
 	});
 	const [totalPages, setTotalPages] = useState(Math.ceil(describeTableData.record_count / pagination.pageSize));
 
-	const { data: tableData, refetch: refetchSearchByValueOptions } = useSuspenseQuery(
+	const { data: tableData, refetch: refetchSearchByValueOptions, isFetching } = useSuspenseQuery(
 		getSearchByValueOptions({
 			instanceId,
 			schemaName,
 			tableName,
 			hash_attribute,
 			sortTableDataParams,
 			pagination,
-		})
+		}),
 	);
+	const { mutate: addTableRecords, isPending: isAddTableRecordsPending } = useInsertTableRecords();
 	const { mutate: updateTableRecords, isPending: isUpdateTableRecordsPending } = useUpdateTableRecords();
 	const { mutate: deleteTableRecords, isPending: isDeleteTableRecordsPending } = useDeleteTableRecords();
 
-	const onRecordUpdate = async (data: object[]) => {
+	useEffect(() => {
+		if (refetchBySearchHash) {
+			void refetchSearchByHash();
+			setSearchByHashParams({ ...searchByHashParams, refetch: false });
+		}
+	}, [refetchBySearchHash, refetchSearchByHash, setSearchByHashParams, searchByHashParams]);
+	useEffect(() => {
+		if (refetchBySortTableParams) {
+			void refetchSearchByValueOptions();
+			setSortTableDataParams({ ...sortTableDataParams, refetch: false });
+		}
+	}, [sortTableDataParams, setSortTableDataParams, refetchBySortTableParams, refetchSearchByValueOptions]);
+	useEffect(() => {
+		setTotalRecords(describeTableData.record_count);
+		setTotalPages(Math.ceil(describeTableData.record_count / pagination.pageSize));
+	}, [
+		describeTableData,
+		pagination.pageSize,
+		pagination.pageIndex,
+	]);
+	useEffect(() => {
+		void refetchSearchByValueOptions();
+	}, [
+		refetchSearchByValueOptions,
+		pagination.pageSize,
+		pagination.pageIndex,
+	]);
```

Comment:
I pulled all of the useEffects to be in one spot, and I broke a few of the larger ones apart a bit. You'll also see a pattern of refetchBySearchHash and refetchBySortTableParams. What that lets us do is trigger a refetch, but wait until the params are updated. It's very possible that we don't need to do this, and we could instead use the query key in the same way, and rely on react-query to only refetch when we're making a new request.